internal/activitypub/person.go

```diff
@@ -34,38 +24,37 @@ type User struct {
 }
 
 func NewUser(
+	id string,
 	name string,
 	matrixUserId string,
 	bio string,
 	publickey string,
-	hostname string,
 ) User {
 	user := User{}
 	user.Context = GetContext()
-	user.Id = fmt.Sprintf("https://%s/u/%s", hostname, name)
+	user.Id = id
 	user.PreferredUsername = name
-	user.Inbox = fmt.Sprintf("https://%s/u/%s/inbox", hostname, name)
-	user.Outbox = fmt.Sprintf("https://%s/u/%s/outbox", hostname, name)
+	user.Inbox = fmt.Sprintf("%s/inbox", id)
+	user.Outbox = fmt.Sprintf("%s/outbox", id)
 	user.Type = "Person"
 	user.Summary = bio
 	user.MatrixUserId = matrixUserId
-	owner := fmt.Sprintf("https://%s/u/%s", hostname, name)
 	user.Publickey = PublicKey{
-		Keyid:        fmt.Sprintf("https://%s/u/%s#main-key", hostname, name),
-		Owner:        owner,
+		Keyid:        fmt.Sprintf("%s#main-key", id),
+		Owner:        id,
 		PublicKeyPem: publickey,
 	}
 	user.Published = time.Now()
-	user.Endpoints.SharedInbox = fmt.Sprintf("https://%s/inbox", hostname)
+	user.Endpoints.SharedInbox = fmt.Sprintf("%s/inbox", id)
 	return user
 }
 
-func ConvertUserToApub(u *lemmy.UserResponse, host string) User {
+func ConvertUserToApub(u *model.Actor) User {
```

Comment:
I think I want to rename these to something more like "ConvertActorToPerson" (and "ConvertActorToGroup") to make slightly more sense of what is actually being changed

Reply:
Except... what would I rename the post one to???

Reply:
Fixed